File: src/msa/web-service/portalapi/src/main/java/vn/com/kns/portalapi/application/service/shop/shopExport/dto/ShopExportDto.java
package vn.com.kns.portalapi.application.service.shop.shopExport.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.com.kns.portalapi.application.service.shop.shop.dto.ShopDto;
import vn.com.kns.portalapi.application.service.shop.shopExportDetail.dto.ShopExportDetailMinDto;
import vn.com.kns.portalapi.core.model.BaseEntityDto;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ShopExportDto extends BaseEntityDto {
    private String code;
    private Long amount;
    private ShopDto shop;
    private Long shopId;
    private List<ShopExportDetailMinDto> shopExportDetails;
}

File: src/msa/main-service/authportal-service/src/app/shared/services/shop/shop-export/dto/shop-export-dto.ts
import {ShopDto} from '../../shop/dto/shop-dto';
import {ShopExportDetailDto} from '../../shop-export-detail/dto/shop-export-detail-dto';

export class ShopExportDto {
  id: number | null;
  code: string;
  amount: number;
  shopId: number;
  shop: ShopDto = new ShopDto();
  shopExportDetails: ShopExportDetailDto[] = [];
}
